manila/share/manager.py

```diff
@@ -4006,10 +4016,14 @@ def delete_snapshot(self, context, snapshot_id, force=False,
                     LOG.exception(msg, snapshot_id)
                     exc.reraise = False
                 else:
+                    error_state = constants.STATUS_ERROR_DELETING
+                    if deferred_delete:
+                        error_state = constants.STATUS_ERROR_DEFERRED_DELETING
+                        exc.reraise = False
```

Comment:
why set `rereaise` to `False`?

Reply:
any state with deferred deleting is hidden from end user. So we dont need to raise exception and handle that error state in periodic task

Reply:
but nothing is changed to the driver, right? Is the deleting API quicker?

Reply:
what is the point of raising exception when we are sure that deferred_deleting and error_deferred_deleting needs to be handled by manila and kept hidden from user. We better move share to error state silently and let periodic function handle it.

Reply:
this would be the process I picture it,

1. share.api sets status of **snapshot instance** to `deleting`
2. share.manager delete_replicated_snapshots
		- delete quota
		- set snapshot to "deleted" and update **instance** status to `defer_deleting`
		- start async job to delete snapshot instances
3. async deleting snapshot job
		- start driver deletion
		- catch all errors, send a message and update instance status to `error_defer_deleting`
		- update instance status to `deleted`

Reply:
> what is the point of raising exception when we are sure that deferred_deleting 

but you are not postponing the deletion, or I miss sth. the logic reaches the exception in the scope of user executing API, isn't it?

Reply:
deferred deletion is now immediate deletion with new states (and those states are hidden from user). The manila will be responsible to delete share/snapshot until its deleted from db. so yes, not postponing deletion. 
the logic reaches in scope of executing API correct. And error_deferred deleting only handled in periodic task

Reply:
> deferred deletion is now immediate deletion with new states (and those states are hidden from user).  

This does not help us to address the slow API problem. If you are not defer deleting, then no need to introduce new states or change the messages. Please reconsider your implementations.

Reply:
this is best we can do as we observed that periodic tasks are shaky. upstream agreed to handle only error_deferred_deleting in periodic task.

we also freeing qouta before deletion making room for new share creation.